### PR TITLE
IDE-122P Add additional tab control for Builder Window and Workunits

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixed IDE-122P

Signed-off-by: dehilsterlexis <david.dehilster@lexisnexis.com>

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for your review

File: eclide/mainfrm.cpp [hunk 1]
```diff
@@ -2685,16 +2692,61 @@ void CMainFrame::DoWorkspaceSwitch(IWorkspace * newWorkspace, bool saveCurrent)
...
+
+    for (POSITION pos = tabbedGroups->GetHeadPosition(); pos != NULL;)
+    {
+        CMFCTabCtrl* tabCtrl = DYNAMIC_DOWNCAST(CMFCTabCtrl, tabbedGroups->GetNext(pos));
+        int activeTab = tabCtrl->GetActiveTab();
+        int total = tabCtrl->GetTabsNum();
+
+        for (int i = 0; i < total; i++)
+        {
+            tabCtrl->GetTabLabel(i, tabstr);
```
Comment (GordonSmith) [inline eclide/mainfrm.cpp]: Can't you compare hwnds rather than display text?

File: en_us/en_us.rc [hunk 2]
```diff
@@ -2083,6 +2083,10 @@ BEGIN
     BEGIN
         MENUITEM "&Save",                       ID_FILE_SAVE
         MENUITEM "&Close\tCtrl+F4",             ID_FILE_CLOSE
+        MENUITEM "Close &All",                  ID_WINDOW_CLOSEALL
+        MENUITEM "Close All &Others",           ID_MDI_CLOSEALLOTHERS
+        MENUITEM "Close All to &Right",         ID_MDI_CLOSEALLRIGHT
```
Comment (GordonSmith) [inline en_us/en_us.rc]: alt+r is already used.

**Comment by GordonSmith** [comment]
Minor comment:  In general I prefer to use "++i" rather than "i++" inside for statements, while it makes no difference for int values, it does have a small performance gain when "i" is an iterator.

File: eclide/mainfrm.cpp [hunk 3]
```diff
@@ -2685,16 +2692,61 @@ void CMainFrame::DoWorkspaceSwitch(IWorkspace * newWorkspace, bool saveCurrent)
...
+            if (tabstr == activetab)
+
```
Comment (GordonSmith) [inline eclide/mainfrm.cpp]: remove blank line

**Comment by dehilsterlexis** [comment]
@GordonSmith I changed the compare to CWnd and also fixed a logical bug where there are multiple tab groups. Back to you.

**Comment by GordonSmith** [comment]
@dehilsterlexis Just the alt+r clash remaining.

**Comment by dehilsterlexis** [comment]
@GordonSmith back to you

**Comment by GordonSmith** [comment]
Looks fine

**Comment by richardkchapman** [comment]
@dehilsterlexis Needs rebasing for a clean merge

**Comment by dehilsterlexis** [comment]
@richardkchapman rebased.

**Comment by GordonSmith** [comment]
@dehilsterlexis Still needs a rebase?  Did you fetch latest from upstream?

**Comment by dehilsterlexis** [comment]
@GordonSmith supposedly I did. I will redo. I had to change my own home on github because my original was forked from my own account so thus the recent confusion.

**Comment by GordonSmith** [comment]
@richardkchapman this can be pulled now?

**Comment by dehilsterlexis** [comment]
@GordonSmith @richardkchapman it is ready.

**Pull request merged**